content/en/docs/How Tos/create_input.md

```diff
@@ -95,3 +95,117 @@ Each field in the struct must include a `help` string tag (mandatory) and a `req
 All these parameters appear in the generated help. `help` should describe the parameter role and/or
 its possible values, `required` informs Baker it should refuse configurations in which that field
 is not defined.
+
+## Write tests
+
+To test an input component we suggest two main paths:
+
+* test the component in isolation, calling the `Run` function
+* test the input at high-level, running a complete Baker topology
+
+Regardless of the chosen path, two additional unit tests are always suggested:
+
+* test the `New()` (constructor-like) function, to check that the function is able to correctly
+instantiate the component with valid configurations and intercept wrong ones
+* create small and isolated functions where possible and unit-test them
+
+### Test calling Run()
+
+In case we want to test the component calling the `Run` function, this is an example of test where,
+after some initialization, the `input.Run` function is called and the produced `Data` is checked
+in a goroutine:
+
+```go
+func TestMyInput(t *testing.T) {
+    ch := make(chan *baker.Data)
+    defer close(ch)
+
+    go func() {
```

Comment:
Maybe we could add a comment saying we start a goroutine that acts as Baker, consuming the baker.Data produced by our tested input?

Reply:
yup, adding